### PR TITLE
refactor: move vanity names to account roles pallet

### PR DESCRIPTION
**Pull request by j4m1ef0rd**
# Pull Request

Closes: PRO-1255

## Checklist

- [x] I am confident that the code works.
- [x] I have updated documentation where appropriate.

## Summary

- Moved the vanity name storage item to the account roles pallet.
	- Moved the related benchmark and test.
- Created a migration. Tested it with try-migration.

### CODE REVIEW

File: state-chain/runtime/src/lib.rs [hunk 1]
```diff
@@ -1039,8 +1038,83 @@ type AllMigrations = (
 	FlipToBurnMigration,
```
Comment (j4m1ef0rd) [inline state-chain/runtime/src/lib.rs:1038]: @dandanlen The `pallet_cf_pools` migrations are commented out but not the `FlipToBurnMigration`, seems strange. I was unable to get my try-migrations to pass without uncommenting the pools pallet migrations. Is this expected?

Comment (dandanlen) [reply state-chain/runtime/src/lib.rs:1038]: Yeah I agree, the pools migrations should not be commented. I'll fix it. 

I want to make some other small improvements.

**Comment by dandanlen** [comment]
I made some additions and resolved conflicts - let me know what you think .

File: state-chain/pallets/cf-account-roles/src/lib.rs [hunk 2]
```diff
@@ -44,11 +53,28 @@ pub mod pallet {
 	#[pallet::storage]
 	pub type AccountRoles<T: Config> = StorageMap<_, Identity, T::AccountId, AccountRole>;
 
+	/// Vanity names of the validators stored as a Map with the current validator IDs as key.
+	#[pallet::storage]
+	#[pallet::getter(fn vanity_names)]
+	pub type VanityNames<T: Config> =
+		StorageValue<_, BTreeMap<T::AccountId, VanityName>, ValueQuery>; // TODO JAMIE: Migration for this
```
Comment (kylezs) [inline state-chain/pallets/cf-account-roles/src/lib.rs:60]:
```suggestion
		StorageValue<_, BTreeMap<T::AccountId, VanityName>, ValueQuery>;
```

**Review by kylezs — APPROVED** [review]

**Pull request merged**